TestingHomework.Tests/DataPrep/PromoCodePrep.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using TestingHomework.Tests.Data;
+using TestingHomework_Discounts;
+
+namespace TestingHomework.Tests.DataPrep
+{
+       public class PromoCodePrep : TypePrepBase<PromoCode, Data.PromoCodeDTO>
+    {
+        public PromoCodePrep(ITestDataAccessor dataAccessor) : base(dataAccessor)
+        {
+
+        }
+
+        public IEnumerable<PromoCode> CreateManyForPromocodes(int count, PromoCode promoCode)
+        {
+            List<PromoCode> promoCodeList = new List<PromoCode>();
+            for (int i = 0; i < count; i++)
+            {
+                PromoCode productList = Create(promoCode);
+                promoCodeList.Add(promoCode);
+            }
+
+            return promoCodeList;
+        }
+        public override PromoCode Create(PromoCode promoCode = null, bool isPersisted = true)
+        {
+            PromoCode sanitizedPromoCode = promoCode ?? Create();
+            PromoCode promocodeItem = new PromoCode()
+            {
+                Id = sanitizedPromoCode.Id,
+                Product = new Product {
```

Comment:
This is an ideal scenario for adding a dependency on the Product data prep (via dependency injection managed in the aggregate dataprep). It would allow you to avoid this duplication of logic with the ProductPrep class